contracts/helpers/RebalancerFlashloanSwap.sol

```diff
@@ -0,0 +1,171 @@
+    /// @dev This function reverts if the second stablecoin mint gives less than `minAmountOut` of stablecoins
+    /// @dev This function reverts if the swap slippage is higher than `maxSlippage`
+    function adjustYieldExposure(
+        uint256 amountStablecoins,
+        uint8 increase,
+        address collateral,
+        address asset,
+        uint256 minAmountOut,
+        bytes calldata swapCallData
+    ) external {
+        if (!TRANSMUTER.isTrustedSeller(msg.sender)) revert NotTrusted();
+        FLASHLOAN.flashLoan(
+            IERC3156FlashBorrower(address(this)),
+            address(AGTOKEN),
+            amountStablecoins,
+            abi.encode(increase, collateral, asset, minAmountOut, swapCallData)
+        );
+    }
+
+    /**
+     * @notice Set the token transfer address
+     * @param _tokenTransferAddress address of the token transfer contract
+     */
+    function setTokenTransferAddress(address _tokenTransferAddress) external onlyGuardian {
+        tokenTransferAddress = _tokenTransferAddress;
+    }
+
+    /**
+     * @notice Set the swap router
+     * @param _swapRouter address of the swap router
+     */
+    function setSwapRouter(address _swapRouter) external onlyGuardian {
+        swapRouter = _swapRouter;
+    }
+
+    /**
+     * @notice Set the max slippage
+     * @param _maxSlippage max slippage in BPS
+     */
+    function setMaxSlippage(uint32 _maxSlippage) external onlyGuardian {
+        maxSlippage = _maxSlippage;
+    }
+
+    /**
+     * @notice Perform the swap using the router/aggregator
+     * @param callData bytes to call the router/aggregator
+     */
+    function _performRouterSwap(bytes memory callData) internal {
+        (bool success, bytes memory retData) = swapRouter.call(callData);
+
+        if (!success) {
+            if (retData.length != 0) {
+                assembly {
+                    revert(add(32, retData), mload(retData))
+                }
+            }
+            revert SwapError();
+        }
+    }
+
+    /**
+     * @notice Swap token using the router/aggregator
+     * @param tokenIn address of the token to swap
+     * @param tokenOut address of the token to receive
+     * @param callData bytes to call the router/aggregator
+     * @param amount amount of token to swap
+     */
+    function _swap(
+        address tokenIn,
+        address tokenOut,
+        bytes memory callData,
+        uint256 amount
+    ) internal returns (uint256) {
+        uint256 balance = IERC20(tokenOut).balanceOf(address(this));
+        _adjustAllowance(tokenIn, tokenTransferAddress, amount);
+        _performRouterSwap(callData);
+        uint256 amountOut = IERC20(tokenOut).balanceOf(address(this)) - balance;
+        if (amountOut < (amount * (BPS - maxSlippage)) / BPS) {
+            revert SlippageTooHigh();
+        }
+        return amountOut;
+    }
+
+    /// @inheritdoc IERC3156FlashBorrower
+    function onFlashLoan(
```

Comment:
A lot of common code with the other rebalancer, let's also try to reuse most of the code so that we don't do copy pasta. 
We can add onFlashLoan in the parent rebalancer and just add virtual internal function to complete based on the case --> rebalance ERC4626, rebalance swap, etc